### PR TITLE
Catalog-of-Things: Ruby Capstone

### PR DESCRIPTION
**Pull request by ZewdieMc**
# Hello Reviewer! 👋🏾 

### Here is a concise list of tasks we have completed :-

- Set up Gitflow correctly
- Used a linter to ensure there are no errors
- Documented work in a professional manner

### Ruby Requirements:

- Followed best practices for Ruby

### Project Requirements - Logic:

- Created each class from the UML class diagram in a separate .rb file
- Implemented all 1-to-many relationships and parent-child aggregations between classes
- Defined and set up all properties visible in the UML diagram in the constructor method, except for properties for 1-to-many relationships, which had a custom setter method created
- Implemented all methods visible in the UML diagram, including:

    -     add_item method in all classes that have it
    -     can_be_archived?() in the Item class
    -     move_to_archive() in the Item class
    -     can_be_archived?() in the Book class (override from parent class)
    -     can_be_archived?() in the MusicAlbum class (override from parent class)
    -     can_be_archived?() in the Movie class (override from parent class)
    -     can_be_archived?() in the Game class (override from parent class)
    -     Added unit tests for all implemented methods
    -     Project Requirements - User Interface:

- Created a main.rb file as the console app entry-point
- Presented a list of options to the user at the start of the app and let them choose an option
- If needed, asked for parameters for the chosen option
- Had a way to quit the app
- Implemented the following options:

    -     Listed all books
    -     Listed all music albums
    -     Listed all games
    -     Listed all genres
    -     Listed all labels
    -     Listed all authors
    -     Added a book
    -     Added a music album
    -     Added a game

- Preserved all data by saving collections in .json files

### Project Requirements - Database:

- Created a schema.sql file with tables that mimic the structure of the classes in the app
- Added all properties and associations from the parent Item class as table columns to all tables based on the child classes

### Project Documentation:

- Recorded a [video](https://drive.google.com/file/d/1NFQ_JmTiCCSLpqZ6EZKZMCjLVDpkTFC2/view) (maximum length of 5 minutes) presenting the features of the project
- Included a description of the project and a demo of its features
- Highlighted an interesting piece of code or something we were proud of
- All team members appeared in the video and talked about the project

### Workload Distribution:

- Divided group tasks among team members or completed them as a team
- Developed other tasks individually based on the Kanban board template
- Created tasks on the Kanban board and worked on them accordingly

### Submission:

- Completed all tasks in one repository as a group

### CODE REVIEW

File: item.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  def move_to_archive
+    @archived = true if can_be_archived?
+  end
+
+  private
+
+  def can_be_archived?
+    Date.today.year - @publish_date.year >= 10
+  end
```
Comment (tugsem) [inline item.rb:26]: - [ ] Please change this method to only return true if the date is older than 10 years as per the requirement.

File: choice.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    input = gets.chomp.to_i
+    if OPTIONS[input]
+      send(OPTIONS[input])
+    else
+      puts 'Invalid input'
+    end
+  end
+
+  def list_books
+    puts '1 - List all books'
```
Comment (tugsem) [inline choice.rb:34]: - [ ] The "list all books" and "add a book" methods actually not calling any methods. Please fix this issue.
image
image

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello @ZewdieMc , @Besufekad-HAZ  and @byiringiroscar 👋 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️great video presentation
✔️all unit tests are passing
✔️readme is professional and descriptive
✔️ app almost works as expected
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by tugsem** [comment]
- [ ] Also the "list all labels" is not working. Kindly add the right methods to list them:  
image

**Comment by Besufekad-HAZ** [comment]
Greetings...

Thanks for the review we've updated that it just a matter or merging branch
and some unresolved bugs.

You can check it in a re-review.

**Review by mwanawabangona — APPROVED** [review]
Hi @Team!!,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implementing the requested chnages !! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**